tests/app/dao/test_provider_details_dao.py

```diff
@@ -155,56 +223,81 @@ def test_reduce_sms_provider_priority_switches_provider(
     # switch away from mmg. currently both 50/50
     dao_reduce_sms_provider_priority('mmg', time_threshold=timedelta(minutes=10))
 
-    assert firetext.priority == expected_priorities['firetext']
-    assert mmg.priority == expected_priorities['mmg']
-    assert mmg.created_by is sample_user
-    assert firetext.created_by is sample_user
+    mock_adjust.assert_any_call(firetext, expected_priorities['firetext'])
+    mock_adjust.assert_any_call(mmg, expected_priorities['mmg'])
 
 
-def test_reduce_sms_provider_priority_adds_rows_to_history_table(
+def test_reduce_sms_provider_priority_does_nothing_if_providers_have_recently_changed(
     mocker,
     restore_provider_details,
-    sample_user
 ):
-    mocker.patch('app.dao.provider_details_dao.get_user_by_id', return_value=sample_user)
-    mmg = get_provider_details_by_identifier('mmg')
-    # need to update these manually to avoid triggering the `onupdate` clause of the updated_at column
-    ProviderDetails.query.filter(ProviderDetails.notification_type == 'sms').update({'updated_at': datetime.min})
+    mock_get_providers = mocker.patch('app.dao.provider_details_dao._get_sms_providers_for_update', return_value=None)
+    mock_adjust = mocker.patch('app.dao.provider_details_dao._adjust_provider_priority')
 
-    provider_history_rows = ProviderDetailsHistory.query.filter(
-        ProviderDetailsHistory.id == mmg.id
-    ).order_by(
-        desc(ProviderDetailsHistory.version)
-    ).all()
+    dao_reduce_sms_provider_priority('firetext', time_threshold=timedelta(minutes=5))
 
-    dao_reduce_sms_provider_priority(mmg.identifier, time_threshold=timedelta(minutes=10))
+    mock_get_providers.assert_called_once_with(timedelta(minutes=5))
+    assert mock_adjust.called is False
 
-    updated_provider_history_rows = ProviderDetailsHistory.query.filter(
-        ProviderDetailsHistory.id == mmg.id
-    ).order_by(
-        desc(ProviderDetailsHistory.version)
-    ).all()
 
-    assert len(updated_provider_history_rows) - len(provider_history_rows) == 1
-    assert updated_provider_history_rows[0].version - provider_history_rows[0].version == 1
-    assert updated_provider_history_rows[0].priority == 90
+@pytest.mark.parametrize('existing_mmg, existing_firetext, new_mmg, new_firetext', [
+    (50, 50, 60, 40),  # not just 50/50 - 60/40 specifically
+    (65, 35, 60, 40),  # doesn't overshoot if there's less than 10 difference
+    (0, 100, 10, 90),  # only adjusts by 10
+    (100, 100, 90, 90),  # it tries to fix weird data
```

Comment:
Optional: Bit unclear what this test case is trying to test as the result new priorities haven't changed from the existing ones

Reply:
i've tried to clarify the comment a little bit - i wanted to prove that the values are independent and it doesn't just bump one and lower the other all the time